src/wl_init.c

```diff
@@ -752,12 +816,91 @@ static void dataDeviceHandleMotion(void* data,
 static void dataDeviceHandleDrop(void* data,
                                  struct wl_data_device* dataDevice)
 {
+    int ret;
+    int fds[2];
+    size_t len = 0;
+    size_t size = 2048;
+    char* buffer = 0;
+
+    if(_glfw.wl.dragDataOffer)
+    {
+        ret = pipe2(fds, O_CLOEXEC);
```

Comment:
I'm not really one to approve or reject PR's, but I wanted to comment:

Does this really need to be `O_CLOEXEC` (are we execing into anywhere?)

Without it the `_GNU_SOURCE` would not be required and this would be more
standards-compliant.

Reply:
This code is based on the reading of the pipe for clipboard data in _glfwGetClipboardStringWayland in wl_window.c. There the O_CLOEXEC flag is also specified. So as this is already a requirement (at least for the wayland backend) I think it is best to also include the flag here.

Reply:
Hmm, you are right. It also seems that both FreeBSD and OpenBSD seem to implement `pipe2` with `O_CLOEXEC` too so it shouldn't really be a future issue either.